web/src/main/kotlin/com/example/ajaxproject/service/PrivateChatServiceImpl.kt

```diff
@@ -13,79 +13,82 @@ import org.bson.types.ObjectId
+                            recipientId = recipient.id
+                        )
+                    )
+                }
             )
     }
 
-    fun roomIdFormat(id1: String, id2: String): String {
-        val smallerId = minOf(id1, id2)
-        val higherId = maxOf(id1, id2)
-        return "$smallerId-$higherId"
-    }
 
-    override fun getPrivateRoom(roomId: String): PrivateChatRoom? {
-        return privateChatRoomRepository.findChatRoomById(roomId) ?:
-            throw NotFoundException("Room with $roomId not fount")
+    override fun getPrivateRoom(roomId: String): Mono<PrivateChatRoom> {
+        return privateChatRoomRepository.findChatRoomById(roomId)
+            .switchIfEmpty(Mono.error(NotFoundException("Room with $roomId not found")))
     }
 
-    override fun sendPrivateMessage(privateMessageDTO: PrivateMessageDTO): PrivateChatMessage {
-        val privateChatMessage = PrivateChatMessage(
-            id = ObjectId(),
-            privateChatRoom = createPrivateRoom(privateMessageDTO.senderId, privateMessageDTO.recipientId),
-            message = privateMessageDTO.message,
-            senderId = privateMessageDTO.senderId
-        )
-        privateChatMessageRepository.save(privateChatMessage)
-
-        logger.info("Send private message {}", privateChatMessage.id)
-
-        return privateChatMessage
+    override fun sendPrivateMessage(privateMessageDTO: PrivateMessageDTO): Mono<PrivateChatMessage> {
+        return createPrivateRoom(privateMessageDTO.senderId, privateMessageDTO.recipientId)
+            .flatMap { room ->
+                val privateChatMessage = PrivateChatMessage(
+                    id = ObjectId(),
+                    privateChatRoomId = room.id,
+                    message = privateMessageDTO.message,
+                    senderId = privateMessageDTO.senderId
+                )
+                privateChatMessageRepository.save(privateChatMessage)
+                    .doOnSuccess { logger.info("Send private message {}", it.id) }
+            }
     }
 
-    override fun getAllPrivateMessages(roomDTO: RoomDTO): List<PrivateChatMessage> {
-
-        userService.getUserById(roomDTO.senderId)
-        userService.getUserById(roomDTO.recipientId)
-
-        val listOfMessage = privateChatMessageRepository.findAllByPrivateChatRoomId(
+    override fun getAllPrivateMessages(roomDTO: RoomDTO): Mono<List<PrivateChatMessage>> {
+        return privateChatMessageRepository.findAllByPrivateChatRoomId(
             roomIdFormat(roomDTO.senderId, roomDTO.recipientId)
         )
+            .collectList()
+            .flatMap { listOfMessage ->
+                if (listOfMessage.isEmpty()) {
+                    logger.warn("Empty chat list")
+                    Mono.error(NotFoundException("Message between users not found"))
+                } else {
+                    logger.info("List of all messages for chat")
+                    Mono.just(listOfMessage)
+                }
+            }
```

Comment:
Alternative: 
```suggestion
            .switchIfEmpty(
                Flux.error { NotFoundException("Message between users not found") }
            )
            .doOnComplete { 
                logger.info("List of all messages for chat") 
            }
```

So, we can return Flux instead of Mono<List>

Reply:
Note! 
```
logger.info("List of all messages for chat") 
```
This log message looks useless

Reply:
Yes , its better to remove them